SpatialGDK/Source/SpatialGDKTests/SpatialGDK/LoadBalanceEnforcer/SpatialLoadBalanceEnforcerTest.cpp

```diff
@@ -0,0 +1,256 @@
+#include "Tests/TestingComponentViewHelpers.h"
+#include "Tests/TestingSchemaHelpers.h"
+
+#include "CoreMinimal.h"
+
+#define LOADBALANCEENFORCER_TEST(TestName) \
+	GDK_TEST(Core, SpatialLoadBalanceEnforcer, TestName)
+
+// Test Globals
+namespace
+{
+
+PhysicalWorkerName ValidWorkerOne = TEXT("ValidWorkerOne");
+PhysicalWorkerName ValidWorkerTwo = TEXT("ValidWorkerTwo");
+
+void AddEntityToStaticComponentView(USpatialStaticComponentView& StaticComponentView,
+	const Worker_EntityId EntityId, VirtualWorkerId Id, Worker_Authority AuthorityIntentAuthority)
+{
+	TestingComponentViewHelpers::AddEntityComponentToStaticComponentView(StaticComponentView,
+		EntityId, SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID,
+		AuthorityIntentAuthority);
+
+	TestingComponentViewHelpers::AddEntityComponentToStaticComponentView(StaticComponentView,
+		EntityId, SpatialConstants::ENTITY_ACL_COMPONENT_ID,
+		WORKER_AUTHORITY_AUTHORITATIVE);
+
+	if (Id >= 0)
+	{
+		SpatialGDK::AuthorityIntent* AuthorityIntentComponent = StaticComponentView.GetComponentData<SpatialGDK::AuthorityIntent>(EntityId);
+		AuthorityIntentComponent->VirtualWorkerId = Id;
+	}
+}
+
+TUniquePtr<SpatialVirtualWorkerTranslator> CreateVirtualWorkerTranslator(bool bAddDefaultMapping = true)
+{
+	ULBStrategyStub* LoadBalanceStrategy = NewObject<ULBStrategyStub>();
+	TUniquePtr<SpatialVirtualWorkerTranslator> VirtualWorkerTranslator = MakeUnique<SpatialVirtualWorkerTranslator>(LoadBalanceStrategy, ValidWorkerOne);
+
+	Schema_Object* DataObject = TestingSchemaHelpers::CreateTranslationComponentDataFields();
+
+	if (bAddDefaultMapping)
+	{
+		TestingSchemaHelpers::AddTranslationComponentDataMapping(DataObject, 1, ValidWorkerOne);
+		TestingSchemaHelpers::AddTranslationComponentDataMapping(DataObject, 2, ValidWorkerTwo);
+	}
+
+	VirtualWorkerTranslator->ApplyVirtualWorkerManagerData(DataObject);
+
+	return VirtualWorkerTranslator;
+}
+
+} // anonymous namespace
+
+LOADBALANCEENFORCER_TEST(GIVEN_load_balance_enforcer_with_no_mapping_WHEN_asked_for_acl_assignments_THEN_return_no_acl_assignment_requests)
+{
+	// This will create a virtual worker translator with no virtual to physical workerId mapping.
+	// This mean the load balance enforcer will not be able to find the physical workerId from entities AuthorityIntent components and therefore fail to produce ACL requests.
+	TUniquePtr<SpatialVirtualWorkerTranslator>  VirtualWorkerTranslator = CreateVirtualWorkerTranslator(false);
+
+	USpatialStaticComponentView* StaticComponentView = NewObject<USpatialStaticComponentView>();
+	AddEntityToStaticComponentView(*StaticComponentView, 0, 1, WORKER_AUTHORITY_AUTHORITATIVE);
+	AddEntityToStaticComponentView(*StaticComponentView, 1, 2, WORKER_AUTHORITY_AUTHORITATIVE);
+
+	TUniquePtr<SpatialLoadBalanceEnforcer> LoadBalanceEnforcer = MakeUnique<SpatialLoadBalanceEnforcer>(ValidWorkerOne, StaticComponentView, VirtualWorkerTranslator.Get());
+
+	LoadBalanceEnforcer->QueueAclAssignmentRequest(0);
```

Comment:
I'd prefer if there's a constant defined with `0` and `1` values and it's passed to `QueueAclAssignmentRequest` instead of hard-coded values.

Reply:
There are multiple places with hard-coded `EntityId` and `WorkerId` values, that may be changed to constants instead. What are your thoughts on this?

Reply:
Yeah sure.